Lib/asyncio/proactor_events.py

```diff
@@ -257,12 +280,24 @@ def _loop_reading__data_received(self, fut):
                 data = None
                 return
 
+            if data is not None:
+                if self._paused:
+                    assert self._pending_data is None
+                    self._pending_data = data
+                else:
+                    self._data_received(data)
```

Comment:
I moved this code because it was easier for me, to be able to implement the switch from data_received to buffer_updated: "if self._protocol_use_buffer(): self._loop_reading__get_buffer(None)" below. I'm not sure if moving this code changes the semantics? It shouldn't, but 3 proactor tests are failing.